ddtrace/tracer/remote_config.go

```diff
@@ -236,6 +285,44 @@ func (t *tracer) onRemoteConfigUpdate(u remoteconfig.ProductUpdate) map[string]s
 	return statuses
 }
 
+// Handle enabling or disabling of Dynamic Instrumentation / Live Debugger.
+//
+// Returns a telemetry configuration if the value changed.
+func (t *tracer) handleDynamicInstrumentationEnabledRC(val *bool) *telemetry.Configuration {
+	// Do not overwrite a "false" value coming from the
+	// DD_DYNAMIC_INSTRUMENTATION_ENABLED env var, or from other local sources.
+	if !t.config.dynamicInstrumentationEnabled.get() &&
+		t.config.dynamicInstrumentationEnabled.cfgOrigin != telemetry.OriginDefault &&
+		t.config.dynamicInstrumentationEnabled.cfgOrigin != telemetry.OriginRemoteConfig {
```

Comment:
I'd actually recommend an explicit allowlist over a disallow list. For a few reasons:
1. There is an origin called "OriginCalculated" which functions similarly to default -- it is a value we've assigned to a configuration based on other factors. So any configuration with a "calculated" origin was not explicitly user configured. I imagine you want to overwrite that case.
2. While there are no plans to do so right now, it is within the realm of possibility that more origins will be added in the future.